keps/sig-network/784-kube-proxy-component-config/README.md

```diff
@@ -127,6 +131,33 @@ The mitigations to those risks:
 
 ## Design Details
 
+### Changes in structure
+
+| v1alpha1               | v1alpha2                | Comments                                                                    |
+|------------------------|-------------------------|-----------------------------------------------------------------------------|
+|                        | Linux                   | new section for linux (platform-specific) options                           |
+| Conntrack              | Linux.Conntrack         | moved from root to linux (platform-specific) section                        |
+| OOMScoreAdj            | Linux.OOMScoreAdj       | moved from root to linux (platform-specific) section                        |
+| IPTables.MasqueradeBit | Linux.MasqueradeBit     | moved from iptables (backend-specific) to linux (platform-specific) section |
+| NFTables.MasqueradeBit | Linux.MasqueradeBit     | moved from nftables (backend-specific) to linux (platform-specific) section |
+| IPTables.MasqueradeAll | Linux.MasqueradeAll     | moved from iptables (backend-specific) to linux (platform-specific) section |
+| NFTables.MasqueradeAll | Linux.MasqueradeAll     | moved from nftables (backend-specific) to linux (platform-specific) section |
+|                        | Windows                 | new section for windows (platform-specific) options                         | 
+| IPTables.SyncPeriod    | SyncPeriod              | moved from iptables (backend-specific) to root (generic) section            |           
+| NFTables.SyncPeriod    | SyncPeriod              | moved from nftables (backend-specific) to root (generic) section            |
+| IPVS.SyncPeriod        | SyncPeriod              | moved from ipvs (backend-specific) to to root (generic) section             |
+| IPTables.MinSyncPeriod | MinSyncPeriod           | moved from iptables (backend-specific) to root (generic) section            |
+| NFTables.MinSyncPeriod | MinSyncPeriod           | moved from nftables (backend-specific) to root (generic) section            |
+| IPVS.MinSyncPeriod     | MinSyncPeriod           | moved from ipvs (backend-specific) to root (generic) section                |
+| ClusterCIDR            | DetectLocal.ClusterCIDR | moved from root to detect-local section                                     |
+| PortRange              |                         | dropped                                                                     |
+
+### Changes in default values
+
+| Field                       | v1alpha1 (default) | v1alpha2 (default) | 
+|-----------------------------|--------------------|--------------------|
+| IPTables.LocalhostNodePorts | true               | false              |
+
 
```

Comment:
Two more sections:

We want to rename some things; especially `BindAddress`, which should be `NodeIPOverride`.

Also, the `HealthzBindAddress` and `MetricsBindAddress` fields are inherently single-stack and need fixing. I think changing them to `HealthzBindInterface` and `MetricsBindInterface` would make sense (though that requires also splitting out separate `HealthzPort` and `MetricsPort`).

Reply:
Should I create a new type `SocketAddress` which encapsulates IP address and port or maybe metav1.SocketAddress ?

We can merge `HealthzBindAddress ` and `HealthzPort` and use list of `SocketAdress` then.

Reply:
`MetricsBindAddress` / `HealthzBindAddress` are already "IP:port", so for what you're suggested we would just make them `[]string` rather than `string` (and then validate them to be either empty, a single value, or a dual-stack pair).

That's definitely simple, but I'm not sure it's the best approach.

The current approach of explicitly specifying an IP makes it hard to say "metrics should be exposed on the management network but not on the network with the default route" because you have to specify a different IP on each node. Whereas if you specify an interface instead, then, hopefully, you can just say "metrics should be exposed on `eth1` only". And, also, specifying an interface is (potentially) dual-stack automatically.

I guess maybe the more-kube-proxy-like approach would be to copy what we did for `nodePortAddresses` and let you specify a CIDR, and then find a local IP within that CIDR... Then we'd definitely need separate port variables though, because there's no standard syntax for "CIDR + port". But the separate port variables might be good anyway; for most people, there's no reason to override the health/metrics _ports_ (and, eg, the GCE loadbalancers won't even work right if you override the health port), so letting people override the metrics/health IPs without having to manually re-specify the unchanged port value is useful...